File: CustomerProfileBank.UI/src/app/models/Role.ts
import { User } from "./User";
import { Privilege } from "./Privilege";

export class Role{
    Id:number;
    Name:string;
    Description?:string;
    Status:string;
    CreationDate:Date;
    Users?:User[]=[];
    Privileges?:Privilege[]=[];
    constructor(){
        this.Name="";
        this.Description="";
        this.Status="Active";
        this.CreationDate=new Date();
        this.Users=[];
        this.Privileges=[];
    }
}

File: CustomerProfileBank.Models/Migrations/Models/Role.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CustomerProfileBank.Models.Models
{
    public class Role
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(50)]
        [Index(IsUnique = true)]
        public string Name { get; set; }

        [Required]
        [MaxLength(500)]
        public string Description { get; set; }

        [Required]
        [MaxLength(50)]
        public string Status { get; set; }
        public virtual ICollection<User> Users { get; set; }
        public virtual ICollection<Privilege> Privileges { get; set; }
        public Role()
        {
            Status = "Active";
        }

        ~Role() { }
    }
}
